### PR TITLE
feat: Add `calculatations_v1` to data product `wholesale_results`

### PR DESCRIPTION
**Pull request by BjarkeMeier**
# Description

This new view `calculations_v1` can be used to obtain information about calculations. In particular there are use cases that need to check whether calculations are internal or external.

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/contracts/data_products/wholesale_results/calculations_v1.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        t.StructField("calculation_period_start", t.TimestampType(), False),
+        #
+        # UTC time. The time is exclusive.
+        t.StructField("calculation_period_end", t.TimestampType(), False),
+        #
+        # Number series per calculation type. Starts from number 1.
+        t.StructField("calculation_version", t.LongType(), False),
+        #
+        # Boolean flag to indicate if the calculation is a control calculation
+        t.StructField("is_internal_calculation", t.BooleanType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/contracts/data_products/wholesale_results/calculations_v1.py:24]: I'm wondering if calculation_completed_time should be included? But I think it is better to leave it out until there is a use case that needs it?

File: source/databricks/calculation_engine/tests/calculator_job/test_energy_calculation.py [hunk 2]
```diff
@@ -239,6 +239,10 @@ def test__when_energy_calculation__calculation_grid_areas_are_stored(
             f"{paths.WholesaleSettlementReportsDatabase.DATABASE_NAME}.{paths.WholesaleSettlementReportsDatabase.ENERGY_PER_ES_VIEW_NAME}",
             True,
         ),
+        (
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculator_job/test_energy_calculation.py:242]: what about test_wholesale_calculation?

File: source/databricks/calculation_engine/contracts/data_products/wholesale_results/calculations_v1.py (hunk 1 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/contracts/data_products/wholesale_results/calculations_v1.py:24]: It's up to you. I considered it as well. But a good colleague of mine once convinced me to expose as little as possible. Because it made it easier to maintain the system and to reason about what is actually being used by others 😉

Comment (JesperGomes) [reply source/databricks/calculation_engine/contracts/data_products/wholesale_results/calculations_v1.py:24]: That's sounds like a clever colleague - I agree with him 😆

File: source/databricks/calculation_engine/tests/calculator_job/test_energy_calculation.py (hunk 2 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/tests/calculator_job/test_energy_calculation.py:242]: Oh, how did I forget that one? Thx.

**Review by JesperGomes — APPROVED** [review]

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161015_create__wholesale_results__calculations_v1.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+DROP VIEW IF EXISTS {CATALOG_NAME}.{WHOLESALE_RESULTS_DATABASE_NAME}.calculations_v1
+GO
+
+CREATE VIEW {CATALOG_NAME}.{WHOLESALE_RESULTS_DATABASE_NAME}.calculations_v1 AS
+SELECT calculation_id,
+       calculation_type,
+       calculation_period_start,
+       calculation_period_end,
+       calculation_version,
+       is_internal_calculation
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161015_create__wholesale_results__calculations_v1.sql:10]: maybe you should wait for my PR and then add 
WHERE calculation_completed_time IS NOT NULL?

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161015_create__wholesale_results__calculations_v1.sql:10]: Man it's hard not to merge when I can! But ok, I'll wait for your PR and add it 😄

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql:1]: Consider adding a view feature test in this PR

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql:1]: ...including the case where succeeded time is NULL?

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+DROP VIEW IF EXISTS {CATALOG_NAME}.{WHOLESALE_RESULTS_DATABASE_NAME}.calculations_v1
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql:1]: why is this needed? Is it because you have deployed on a dev environment? If yes, you can just clean up that environment manually.

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql:1]: It's about being pragmatic about local development: Avoiding to need to rerun all migrations locally.

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql:1]: Adding - and will be fixed as well 🙈

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql (hunk 4 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408161300_create__wholesale_results__calculations_v1.sql]: Turns out I had to rerun all migrations locally anyway. So I've removed it.

**Review by JesperGomes — APPROVED** [review]
